development/extensions/tutorial_notifications.rst

```diff
@@ -0,0 +1,955 @@
+   :local:
+
+Introduction
+============
+Notifications have been a part of phpBB ever since :guilabel:`3.0`.
+Since then they have been further developed, extended and enhanced.
+In :guilabel:`3.2` the notification system has been completely refactored, so it worked faster and more efficiently.
+This change also reduced the posting time to be almost instantaneous.
+
+A user has full power over their personal notifications.
+They can set their own preferences for notification methods or disable them all together.
+Mark notifications as read when necessary, even though in a lot of cases this is done automatically.
+The user has a centralised place for all notifications, which is easily accessible.
+
+By default there are two methods of notification.
+Namely per email or per board notification.
+The email method is only available when the ''board-wide emails'' setting has been enabled.
+Extensions can also create and add their own notification methods.
+
+File structure
+--------------
+This is an overview of the file structure needed for a notification to function within an extension.
+In total there are 7 files that will be covered in this tutorial, to create and send notifications.
+These files will be mentioned throughout this tutorial.
+
+
+| |fa-user| vendor
+| └─ |fa-ext| extension
+|  |nbsp| ├─ |fa-folder| config
+|  |nbsp| │ |nbsp| └─ |fa-file| services.yml
+|  |nbsp| ├─ |fa-folder| event
+|  |nbsp| │ |nbsp| └─ |fa-file| listener.php
+|  |nbsp| ├─ |fa-folder| language
+|  |nbsp| │ |nbsp| |nbsp| └─ |fa-folder| en
+|  |nbsp| │ |nbsp| |nbsp| |nbsp| |nbsp| |nbsp| |nbsp| └─ |fa-folder| email
+|  |nbsp| │ |nbsp| |nbsp| |nbsp| |nbsp| |nbsp| |nbsp| │ |nbsp| |nbsp| └─ |fa-folder| short
+|  |nbsp| │ |nbsp| |nbsp| |nbsp| |nbsp| |nbsp| |nbsp| │ |nbsp| |nbsp| │ |nbsp| |nbsp| └─ |fa-file| sample.txt
+|  |nbsp| │ |nbsp| |nbsp| |nbsp| |nbsp| |nbsp| |nbsp| │ |nbsp| |nbsp| └─ |fa-file| sample.txt
+|  |nbsp| │ |nbsp| |nbsp| |nbsp| |nbsp| |nbsp| |nbsp| └─ |fa-file| extension_common.php
+|  |nbsp| ├─ |fa-folder| notification
+|  |nbsp| │ |nbsp| |nbsp| └─ |fa-folder| type
+|  |nbsp| │ |nbsp| |nbsp| |nbsp| |nbsp| |nbsp| |nbsp| └─ |fa-file| sample.php
+|  |nbsp| └─ |fa-file| ext.php
+|
+
+Creating a notification type
+============================
+The notification type that will be used in this tutorial is: ``vendor.extension.notification.type.sample``.
+
+Registering as a Symfony service
+--------------------------------
+Firstly the notification type has to be registered as a Symfony service.
+As with any service declaration, it will need a service identifier and the class argument.
+However, notifications types also need three additional parameters. The parent, shared and tags parameters.
+The parent parameter will indicate from which service this notification type will inherit.
+And the shared parameter will make sure that each time this notification type is requested, a new instance is created.
+This will prevent that any data will be transferred over.
```

Comment:
This line should be re-worded to make better sense.

Reply:
> This will prevent that any data from one notification is transferred over to an other.

Went with that for now.
Not sure if we should write an entire block on how symfony services are shared between classes. Perhaps if we want to elaborate on it, we could add something like:

> For example, when a user has two notifications, both are our sample notification type.
The *Notification manager object* will retrieve this notification class from the service container. On the first call, a new instance of this notification type class will be created. However, if `shared` is set to `true`, the second time the notification manager tries to retrieve this notification class, the service container will return the instance that was created the first time around. Meaning that the `$data` property still contains the information about the previous notification. Such as the `sender_id`, `message_id`, etc..